modules/govuk/lib/puppet/parser/functions/data_sync_times.rb

```diff
@@ -0,0 +1,38 @@
+module Puppet::Parser::Functions
+  newfunction(:data_sync_times, :type => :rvalue, :doc => <<-EOS
+Return information about the data sync times, depending on the argument passed
+    EOS
```

Comment:
Ooh this is exotic, I've never seen a puppet function definition before. It seems a neat solution to the problem 👍

Reply:
I took inspiration from [other functions we have in govuk-puppet](https://github.com/alphagov/govuk-puppet/blob/f9f16fc8756e3965af6b697175f71d6c94343179/modules/govuk/lib/puppet/parser/functions/monitoring_docs_url.rb), after initially trying (and failing) to define the start/finish times in YAML. Could have pursued YAML further but as we sometimes need the raw values and sometimes need a representation of them, seemed just as clean to put them in here. 👍